app/src/main/java/co/nimblehq/ui/base/BaseNavigator.kt

```diff
@@ -0,0 +1,79 @@
+package co.nimblehq.ui.base
+
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.annotation.IdRes
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavController
+import androidx.navigation.fragment.findNavController
+import co.nimblehq.extension.getResourceName
+import timber.log.Timber
+
+interface BaseNavigator {
+
+    val navHostFragmentId: Int
+
+    fun findNavController(): NavController?
+
+    fun navigate(event: NavigationEvent)
+
+    fun navigateUp()
+}
+
+abstract class BaseNavigatorImpl(protected val fragment: Fragment) : BaseNavigator {
+
+    private var navController: NavController? = null
+
+    override fun findNavController(): NavController? {
+        return navController ?: try {
+            fragment.findNavController()
+        } catch (e: IllegalStateException) {
+            // Log Crashlytics as non-fatal for monitoring
+            Timber.e(e)
+            null
+        }
+    }
+
+    override fun navigateUp() {
+        findNavController()?.navigateUp()
+    }
+
+    protected fun popBackTo(@IdRes destinationId: Int, inclusive: Boolean = false) {
+        findNavController()?.popBackStack(destinationId, inclusive)
+    }
+
+    protected fun unsupportedNavigation() {
+        val navController = findNavController()
+        val currentGraph = fragment.requireActivity().getResourceName(navController?.graph?.id)
+        val currentDestination = fragment.requireActivity().getResourceName(navController?.currentDestination?.id)
+        handleError(NavigationError.UnsupportedNavigationError(currentGraph, currentDestination))
+    }
+
+    protected fun NavController.navigateToDestinationByDeepLink(
+        destinationId: Int,
+        bundle: Parcelable? = null
+    ) {
+        createDeepLink()
+            .setDestination(destinationId)
+            .apply {
+                bundle?.let {
+                    setArguments(
+                        Bundle().apply {
+                            putParcelable("bundle", bundle)
+                        }
+                    )
+                }
+            }
```

Comment:
maybe shorten this part a bit:
```suggestion
            .apply {
                bundle?.let { setArguments(Bundle().apply { putParcelable("bundle", bundle) }) }
            }
```